nntrainer/tensor/manager.cpp

```diff
@@ -698,32 +633,20 @@ void Manager::initializeTensorsTrain() {
       }
     }
   } else {
-    tensor_pool.finalize(BasicPlanner(), 0, max_exec_order);
-
-    // Training Mode without optimizations
-    for (auto &outs : outputs_v2) {
-      outs->initialize(Tensor(), Tensor(), true);
-    }
-
-    // Training Mode without optimizations
-    // for (auto &ts : tensors_v2) {
-    //   ts->initialize(Tensor(), Tensor(), true);
-    // }
-
-    // Training Mode without optimizations
-    for (auto &ins : inputs_v2) {
-      ins->initialize(Tensor(), Tensor(), true);
-    }
+    tensor_pool.finalize(BasicPlanner(), 0, max_exec_order_);
   }
 }
 
 /**
  * @brief Initialize the inputs/outputs/gradients/derivatives for the layer
  */
-void Manager::initializeTensors(bool training) {
+void Manager::initializeTensors(bool training, unsigned int max_exec_order_) {
```

Comment:
What does max_exec_order_ do? is this initialize Until the given number?

Reply:
Yes, this tells the manager till which node is the network going to run. One example to exclude allocation gradients while inference as their execution order would be more than the provided max_exec_order.

Reply:
Cool